crates/rspack_browser/src/oneshot.rs

```diff
@@ -0,0 +1,82 @@
+//! Busy-wait version oneshot channel aiming to be the drop-in replacement of other oneshot crates such [tokio::sync::oneshot]
+//!
+//! When [Receiver] start to receive before [Sender::send], normally the receiver will be put into the task pool and be waken by the [Sender].
+//! However, runtime should acquire the lock of task pool before waking the receiver, which is forbidden in the main thread of browser.
+//! So the busy-wait [Receiver] is provided.
+use std::{
+  cell::UnsafeCell,
+  future::Future,
+  pin::Pin,
+  sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+  },
+  task::{Context, Poll},
+};
+
+pub fn channel<T>() -> (Sender<T>, Receiver<T>) {
+  let inner = Arc::new(Inner {
+    sent: AtomicBool::new(false),
+    value: UnsafeCell::new(None),
+  });
+
+  let tx = Sender {
+    inner: inner.clone(),
+  };
+
+  let rx = Receiver {
+    inner: inner.clone(),
+  };
+
+  (tx, rx)
+}
+
+struct Inner<T> {
+  sent: AtomicBool,
+  value: UnsafeCell<Option<T>>,
+}
+
+unsafe impl<T: Send> Send for Inner<T> {}
+unsafe impl<T: Send> Sync for Inner<T> {}
+
+pub struct Sender<T> {
+  inner: Arc<Inner<T>>,
+}
+
+impl<T> Sender<T> {
```

Comment:
`Drop` should be implement for `Sender`, otherwise the `Receiver` will never end if `Sender` is released.